### PR TITLE
Check via CDX API if URL archived within the last 30 days

### PR DESCRIPTION
**Pull request by atesgoral**
* Changes to the URL field (keyup, paste, change events) trigger the CDX URL lookup with a debounce of 500ms (i.e. 500ms after user's last action)
* We only fetch the timestamp field to reduce payload returned from CDX
* CDX results are normalized into an array of objects with the timestamp properties converted to JS Dates
* Only one result is requested right now (the most recent)
* No visual indication of this background check is given unless the URL is found to be archived within the last 30 days -- in which case a warning is shown
* Consolidated success/error notifications to use Bootstrap's alerts
* Did some miscellaneous refactoring

### CODE REVIEW

File: popup.js [hunk 1]
```diff
@@ -111,28 +143,28 @@ function nominationTool( e ) {
...
-    url: GOOGLE_FORMS_URL,
-    data,
-    success: function( res ) {
-      $( '#success' ).html( "Success!" );
+  $.get(GOOGLE_FORMS_URL, data)
+    .then(() => {
+      $('#save').prop('disabled', false);
+
+      showStatus('success', 'URL submitted. Thanks!');
+      // @todo Why are we reloading the popup here? To reset some fields?
```
Comment (titaniumbones) [inline popup.js]: I'm not sure but I think part of the reason was to just flash the success field. But people often miss the notification, so actually maybe it's not good UI.

**Review by titaniumbones — APPROVED** [review]
Refactoring looks good. I'll just run a couple of tests -- we should have a suite of tests!!! -- but should I merge or wait for the new feature?

**Comment by atesgoral** [comment]
@titaniumbones The feature is already in this PR. I broke my own principle of keeping PRs topical and did the new feature + refactoring in one shot.

File: popup.js (hunk 1 above)
Comment (atesgoral) [reply popup.js]: How about we hide the success field after a short delay?

Comment (titaniumbones) [reply popup.js]: sure.  I was just fooling aorund with it. 1000ms is too short I htink, mroe like 5000?

**Comment by titaniumbones** [comment]
@atesgoral did you see @mhucka 's notes about the 503? Does it give you any concerns? I guess now that you're not getting the 503's it's hard to check the Retry-After header.

**Comment by atesgoral** [comment]
Yeah, it's too late now to check if there was a Retry-After header.

Wishfully thinking that it was just a fluke :)

It would be interesting to hear from the IA folks what they think about the 503 responses.

I'm looking around for a cloud-based API caching gateway. Amazon's API Gateway can be configured to cache API responses from an endpoint for a certain amount of time, but it's not available on the free tier. Also, it's optimistic to think that there will be a lot of cache hits to be able to take advance of an API response cache.

**Comment by atesgoral** [comment]
Realized I forgot to check the URL when the popup initially loads :) Adding that now.

File: popup.js (hunk 1 above)
Comment (atesgoral) [reply popup.js]: 3-5 sec should be good. I'll try it out.

**Comment by mhucka** [comment]
Just a quick note to say I haven't heard anything back yet. Will keep you posted. I'm sure the IA people are beyond busy these days ...

**Comment by atesgoral** [comment]
@mhucka, @titaniumbones It could be best if I just leave the feature code in, but just disable the CDX lookup so that we can at least merge this PR with all the refactoring that was done. Then it would be just flipping a switch to re-enable the lookup.

**Comment by mhucka** [comment]
@atesgoral Sounds like a most excellent plan!

**Comment by atesgoral** [comment]
URL checking disabled.

**Comment by titaniumbones** [comment]
Excellent, merging. @mhucka, if somehow you hear back form Jefferson that the problem is unrelated to server load, we can revert 18e69ee and take this live at any point.

**Comment by titaniumbones** [comment]
Should i delete the branch or keep it open?

**Comment by atesgoral** [comment]
Safe to delete! Deleted.

**Pull request merged**